sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/BlobQueryReader.java

```diff
@@ -230,10 +231,17 @@ public static QuerySerialization transformInputSerialization(BlobQuerySerializat
             generatedFormat.setJsonTextConfiguration(transformJson(
                 (BlobQueryJsonSerialization) userSerialization));
 
+        } else if (userSerialization instanceof BlobQueryParquetSerialization) {
+
+            generatedFormat.setType(QueryFormatType.PARQUET);
+            generatedFormat.setParquetTextConfiguration(transformParquet(
+                (BlobQueryParquetSerialization) userSerialization));
+
         } else {
             throw logger.logExceptionAsError(new IllegalArgumentException(
-                String.format("'input' must be one of %s or %s", BlobQueryJsonSerialization.class.getSimpleName(),
-                    BlobQueryDelimitedSerialization.class.getSimpleName())));
+                String.format("'input' must be one of %s, %s or %s", BlobQueryJsonSerialization.class.getSimpleName(),
```

Comment:
Are the allowed input types expected to continue growing? If so, I think it would be good to create external documentation somewhere and just have this error message point towards that.

Reply:
https://docs.microsoft.com/en-us/rest/api/storageservices/query-blob-contents#request-body

Below the example xml, there is a table explaining each parameter.

Reply:
Just updated the error message. @alzimmermsft can  you take a quick look?